### DISCUSSION TITLE
Spline widget clipping

### DISCUSSION THREAD

**Discussion by NahomKidane** [pyvista/pyvista#4593] [Q&A]
Is there a way to create a spline clipping tool? Similar to the 3DSlicer curve cut under the dynamic modeler. The user will select a region using the spline, and we can clip the inside/outside surface.  (https://slicer.readthedocs.io/en/latest/user_guide/modules/dynamicmodeler.html)

![image](https://github.com/pyvista/pyvista/assets/7750047/39bfddb5-fe21-4da1-af59-371c37910d19)

**Comment by banesullivan** [comment]
Possible, yes! Implemented, not yet... You could get started with the spline widget we do have and a callback to use that widget to perform the cut operation.

https://docs.pyvista.org/version/stable/examples/03-widgets/spline-widget.html

Then you could extrude the spline to create a cut surface and use `clip_surface()` to perform the clip:

https://docs.pyvista.org/version/stable/examples/01-filter/clipping-with-surface.html

This should get you started at least but I suspect the spline widget will need to form a closed loop
